### ISSUE TITLE
Add PMD information to developer docs.

### ISSUE THREAD

**Issue by m273d15** [saros-project/saros#1013]
With the new PMD `ruleset.xml` it is possible to
integrate the necessary static code analysis in the IDE.
Therefore, we could add information on how to run this
analysis locally.

**Comment by m273d15** [comment]
@tobous Do you use the PMD IntelliJ integration?

**Comment by tobous** [comment]
No, I did not look into an IntelliJ integration yet as I haven't worked on a branch where the configuration file exists yet. From having a look at it now, there are [two plugins that support it](https://pmd.github.io/latest/pmd_userdocs_tools.html#idea).

But I am not sure whether such an integration is worth it (at least for me). It seems like you can't integrate it with the normal code inspection but rather have to manually run the analysis for a specific set of resources every time you want to check it (and you also have to run this analysis through a different menu entry added by the plugin that you first have to find).
Manually running PMD is something that I can also do in the CLI before committing (or even run the gradle task from inside IntelliJ). Yes, the plugin provides a nicer overview over existing issues, but, as we want to enforce the enabled rules, there shouldn't be too many issues at any given time anyways.

But, if you want to add documentation about IntelliJ IDEA integration, I would suggest just linking to https://pmd.github.io/latest/pmd_userdocs_tools.html#idea. The page explains how to integrate a normal PMD runnable and lists the available plugins.

**Comment by m273d15** [comment]
> But, if you want to add documentation about IntelliJ IDEA integration

No, I like to keep the docs as small as possible. I will just add the information how to run the checks via gradle.
